osu.Game/Overlays/Mods/BeatmapAttributesDisplay.cs

```diff
@@ -148,6 +148,22 @@ private void startAnimating()
             LeftContent.AutoSizeDuration = Content.AutoSizeDuration = transition_duration;
         }
 
+        protected virtual double GetRate()
+        {
+            double rate = 1;
+            foreach (var mod in Mods.Value.OfType<IApplicableToRate>())
+                rate = mod.ApplyToRate(0, rate);
+            return rate;
+        }
+
+        protected virtual BeatmapDifficulty GetDifficulty()
+        {
+            BeatmapDifficulty originalDifficulty = new BeatmapDifficulty(BeatmapInfo.Value!.Difficulty);
+            foreach (var mod in Mods.Value.OfType<IApplicableToDifficulty>())
+                mod.ApplyToDifficulty(originalDifficulty);
+            return originalDifficulty;
+        }
```

Comment:
This definitely should not be required, though. Changes should stop roughly at the `MultiplayerModSelectOverlay` level.

Reply:
i just moved those lines to separate functions so i can change them in `MultiplayerModSelectOverlay`
i need to either do this or make entire `updateValues` virtual, what will lead to code duplication

Reply:
in initial `updateValues` it's using only Mods list to calculate Rate, but in multiplayer version - it using not only Mods but also RequiredMods

Reply:
You shouldn't need to override the methods that _calculate_ the values. You should override _which set mods these methods use to calculate from_.

Reply:
okay, i get it